Place a Mechanical Pump by other mod will crash the game

### Context

**Expected behavior**
Placing blocks in Create mod via other mods should not crash.
https://github.com/baileyholl/Ars-Nouveau/issues/1824#issuecomment-2963456553

**Observed/Actual behavior**
When using a Ars-Nouveau spell to place a Mechanical Pump in Create mod above, below, or next to (except south of) a Fluid Tank in Create mod, if the spell has a Projectile spell attached, the game will crash immediately. If the Touch spell is attached, it will not crash, but the block will not be placed when the item is consumed. At the same time, the following error message will appear in latest.log:
```
[Server thread/ERROR] [net.minecraft.util.thread.BlockableEventLoop/FATAL]: Error executing task on Server
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "net.minecraft.world.entity.player.Player.isShiftKeyDown()" because "player" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770) ~[?:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.TickTask.run(TickTask.java:18) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:148) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:872) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:170) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:855) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:849) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.managedBlock(MinecraftServer.java:821) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:826) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:712) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:267) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.world.entity.player.Player.isShiftKeyDown()" because "player" is null
	at TRANSFORMER/create@6.0.4/com.simibubi.create.content.fluids.pump.PumpBlock.getStateForPlacement(PumpBlock.java:128) ~[create-1.21.1-6.0.4.jar%23196!/:6.0.4]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.item.BlockItem.getPlacementState(BlockItem.java:131) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.item.BlockItem.place(BlockItem.java:64) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.spell.effect.EffectPlaceBlock.attemptPlace(EffectPlaceBlock.java:95) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.spell.effect.EffectPlaceBlock.place(EffectPlaceBlock.java:84) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.spell.effect.EffectPlaceBlock.onResolveBlock(EffectPlaceBlock.java:69) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.AbstractEffect.onResolve(AbstractEffect.java:57) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.SpellResolver.resume(SpellResolver.java:195) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.SpellResolver.resolveAllEffects(SpellResolver.java:162) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.SpellResolver.onResolveEffect(SpellResolver.java:154) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.spell.method.MethodTouch.onCastOnBlock(MethodTouch.java:44) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.SpellResolver.onCastOnBlock(SpellResolver.java:127) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.AbstractCaster.castSpell(AbstractCaster.java:294) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.network.PacketCastSpell.onServerReceived(PacketCastSpell.java:75) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.network.Networking.handleServer(Networking.java:95) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.network.Networking.handle(Networking.java:85) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/neoforge@21.1.172/net.neoforged.neoforge.network.handling.MainThreadPayloadHandler.lambda$handle$0(MainThreadPayloadHandler.java:16) ~[neoforge-21.1.172-universal.jar%23194!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.lambda$submitAsync$0(BlockableEventLoop.java:60) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
	... 14 more

[Server thread/ERROR] [net.neoforged.neoforge.network.registration.NetworkRegistry/]: Failed to process a synchronized task of the payload: ars_nouveau:cast_spell
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "net.minecraft.world.entity.player.Player.isShiftKeyDown()" because "player" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770) ~[?:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.TickTask.run(TickTask.java:18) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:148) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:872) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:170) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:855) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:849) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.managedBlock(MinecraftServer.java:821) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:826) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:712) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:267) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.world.entity.player.Player.isShiftKeyDown()" because "player" is null
	at TRANSFORMER/create@6.0.4/com.simibubi.create.content.fluids.pump.PumpBlock.getStateForPlacement(PumpBlock.java:128) ~[create-1.21.1-6.0.4.jar%23196!/:6.0.4]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.item.BlockItem.getPlacementState(BlockItem.java:131) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.item.BlockItem.place(BlockItem.java:64) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.spell.effect.EffectPlaceBlock.attemptPlace(EffectPlaceBlock.java:95) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.spell.effect.EffectPlaceBlock.place(EffectPlaceBlock.java:84) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.spell.effect.EffectPlaceBlock.onResolveBlock(EffectPlaceBlock.java:69) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.AbstractEffect.onResolve(AbstractEffect.java:57) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.SpellResolver.resume(SpellResolver.java:195) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.SpellResolver.resolveAllEffects(SpellResolver.java:162) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.SpellResolver.onResolveEffect(SpellResolver.java:154) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.spell.method.MethodTouch.onCastOnBlock(MethodTouch.java:44) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.SpellResolver.onCastOnBlock(SpellResolver.java:127) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.api.spell.AbstractCaster.castSpell(AbstractCaster.java:294) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.network.PacketCastSpell.onServerReceived(PacketCastSpell.java:75) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.network.Networking.handleServer(Networking.java:95) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/ars_nouveau@5.8.3/com.hollingsworth.arsnouveau.common.network.Networking.handle(Networking.java:85) ~[ars_nouveau-1.21.1-5.8.3-all.jar%23195!/:5.8.3]
	at TRANSFORMER/neoforge@21.1.172/net.neoforged.neoforge.network.handling.MainThreadPayloadHandler.lambda$handle$0(MainThreadPayloadHandler.java:16) ~[neoforge-21.1.172-universal.jar%23194!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.lambda$submitAsync$0(BlockableEventLoop.java:60) ~[client-1.21.1-20240808.144430-srg.jar%23193!/:?]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
	... 14 more
```

**Version**
Minecraft 1.21.1 NeoForge 21.1.172
mods: Ars Nouveau 1.21.1-5.8.3, Create 1.21.1-6.0.4

(tested create 6.0.6 and ars_nouveau-1.21.1-5.8.6, and also crashed.)

### Crash Report

https://mclo.gs/6JY6LdX

--- a/src/main/java/com/simibubi/create/content/fluids/pump/PumpBlock.java
+++ b/src/main/java/com/simibubi/create/content/fluids/pump/PumpBlock.java
@@ -99,12 +99,12 @@ public class PumpBlock extends DirectionalKineticBlock
 		BlockState toPlace = super.getStateForPlacement(context);
 		Level level = context.getLevel();
 		BlockPos pos = context.getClickedPos();
-		Player player = context.getPlayer();
+
+		boolean isShiftKeyDown = context.getPlayer() != null && context.getPlayer().isShiftKeyDown();
 		toPlace = ProperWaterloggedBlock.withWater(level, toPlace, pos);
 
 		Direction nearestLookingDirection = context.getNearestLookingDirection();
-		Direction targetDirection = context.getPlayer() != null && context.getPlayer()
-			.isShiftKeyDown() ? nearestLookingDirection : nearestLookingDirection.getOpposite();
+		Direction targetDirection = isShiftKeyDown ? nearestLookingDirection : nearestLookingDirection.getOpposite();
 		Direction bestConnectedDirection = null;
 		double bestDistance = Double.MAX_VALUE;
 
@@ -121,14 +121,10 @@ public class PumpBlock extends DirectionalKineticBlock
 			bestConnectedDirection = d;
 		}
 
-		if (bestConnectedDirection == null)
-			return toPlace;
-		if (bestConnectedDirection.getAxis() == targetDirection.getAxis())
-			return toPlace;
-		if (player.isShiftKeyDown() && bestConnectedDirection.getAxis() != targetDirection.getAxis())
-			return toPlace;
+		if (bestConnectedDirection != null && bestConnectedDirection.getAxis() != targetDirection.getAxis() && !isShiftKeyDown)
+			return toPlace.setValue(FACING, bestConnectedDirection);
 
-		return toPlace.setValue(FACING, bestConnectedDirection);
+		return toPlace;
 	}
 
 	public static boolean isPump(BlockState state) {

